### PR TITLE
New vanilla website

### PR DESCRIPTION
**Pull request by albertkol**
## Done

- Remove old "Design" section from docs
- Extract Accessibility docs into their own file
- Rename Tab buttons to segmented control
- Add tab menu on the docs pages for accessibility and design guidelines.
- Docs tab menu settings are set on `component_tabs.yaml` file.

## QA

- Code has already been reviewed for each PR separately - this PR is the final QA
- Go to /docs check the tab menus for each pattern and layout

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

## Screenshots

image

### CODE REVIEW

**Comment by bethcollins92** [comment]
Noticed in QA - on the design guidelines page, the subtitles don't appear in the side nav

File: templates/_layouts/docs_discourse.html [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 {% block content %}
 <h1>{{ document.title }}</h1>
```
Comment (lyubomir-popov) [inline templates/_layouts/docs_discourse.html:6]:
```suggestion
<div class="p-strip u-no-padding--top">
<h1>{{ document.title }}</h1>
</div>
```

Comment (lyubomir-popov) [reply templates/_layouts/docs_discourse.html:6]: @albertkol small suggestion

**Review by lyubomir-popov — DISMISSED** [review]

**Comment by lyubomir-popov** [comment]
Hi @bethcollins92, my proposal here was to decrease the space above, increase the space between the heading, to create a more deliberate look:

image

**Comment by bethcollins92** [comment]
Thanks @lyubomir-popov - made those changes

**Comment by lyubomir-popov** [comment]
Thanks @beth, looks good.

**Comment by bethcollins92** [comment]
Woop. @bartaz this is ready for final QA check now

**Review by bartaz — CHANGES_REQUESTED** [review]
There are some pages with missing title:
https://vanilla-framework-4430.demos.haus/docs/patterns/contextual-menu
https://vanilla-framework-4430.demos.haus/docs/patterns/icons

maybe others as well.

We need to find out why and fix it.

File: component_tabs.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+patterns/accordion:
+  accessibility: True
+patterns/breadcrumbs:
+  accessibility: True
+patterns/buttons:
+  accessibility: True
+patterns/card:
+  accessibility: True
+patterns/contextual-menu:
```
Comment (bartaz) [inline component_tabs.yaml:11]: Should this one be empty? It renders incorrectly without title at all.

I don't know if it's supposed to have tabs or not. If not I guess it shouldn't be in the file at all?

**Comment by bartaz** [comment]
Segmented control page has missing `<hr>` under the title

**Comment by bartaz** [comment]
It feels weird to have more spacing on pages with tabs and no spacing on old docs without tabs. I think we should add the space to old templates as well. I guess it may end up a bigger change, not sure if we need to put it in this PR if it's too large.

docs-title-strip

File: component_tabs.yaml (hunk 2 above)
Comment (bethcollins92) [reply component_tabs.yaml:11]: We haven't updated the accessibility docs for this page, so currently there is a short paragraph:

"When using the `p-contextual-menu__toggle` class on a `button` element, please ensure that the button label contains a trailing ellipsis `…`, e.g. "Take action&hellip;". This is a convention used to indicate that the button launches a dialog.

In cases where a contextual menu is shown on click, focus should be set within the menu element, using JavaScript."

It feels like overkill to have a tab for this, so I'm leaning towards putting the text back into the main file and having no tabs for this page until we have complete accessibility docs

Comment (bartaz) [reply component_tabs.yaml:11]: Sounds good.

File: component_tabs.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+patterns/grid:
+  accessibility: True
+patterns/heading-icon:
+  accessibility: True
+patterns/icon:
```
[the remark this answers is not in the document]
Comment (bethcollins92) [reply component_tabs.yaml]: I believe it was - fixed now

**Comment by bethcollins92** [comment]
I've fixed:

- Pages with missing titles
- `<hr>` added under segmentel control
- An issue is created [here](https://github.com/canonical-web-and-design/vanilla-framework/issues/4440) about the spacing under headings issue

**Review by bartaz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**